src/OrbitGl/TimeGraph.cpp

```diff
@@ -455,52 +457,37 @@ void TimeGraph::ProcessSystemMemoryTrackingTimer(const TimerInfo& timer_info) {
     return;
   }
 
-  constexpr double kMegabytesToKilobytes = 1024.0;
-  orbit_gl::SystemMemoryTrack* track = track_manager_->GetSystemMemoryTrack();
+  SystemMemoryTrack* track = track_manager_->GetSystemMemoryTrack();
   if (track == nullptr) {
-    const std::array<std::string, orbit_gl::kSystemMemoryTrackDimension> kSeriesNames = {
-        "Used", "Buffers / Cached", "Unused"};
-    track = track_manager_->CreateAndGetSystemMemoryTrack(kSeriesNames);
+    track = track_manager_->CreateAndGetSystemMemoryTrack();
   }
+
+  constexpr double kMegabytesToKilobytes = 1024.0;
   CHECK(track->GetNumberOfDecimalDigits().has_value());
   uint8_t num_of_decimal_digits = track->GetNumberOfDecimalDigits().value();
+  double total_mb =
+      RoundPrecision(static_cast<double>(total_kb) / kMegabytesToKilobytes, num_of_decimal_digits);
   double unused_mb =
       RoundPrecision(static_cast<double>(unused_kb) / kMegabytesToKilobytes, num_of_decimal_digits);
   double buffers_or_cached_mb = RoundPrecision(
       static_cast<double>(buffers_kb + cached_kb) / kMegabytesToKilobytes, num_of_decimal_digits);
-  double used_mb =
-      RoundPrecision(static_cast<double>(total_kb) / kMegabytesToKilobytes, num_of_decimal_digits) -
-      unused_mb - buffers_or_cached_mb;
-  track->AddValues(timer_info.start(), {used_mb, buffers_or_cached_mb, unused_mb});
+  double used_mb = total_mb - unused_mb - buffers_or_cached_mb;
+  std::array<double, orbit_gl::kSystemMemoryTrackDimension> values;
+  values[static_cast<size_t>(SystemMemoryTrack::SeriesIndex::kUsedMb)] = used_mb;
```

Comment:
This is because the we are using array to record values at the same timestamp in `MultivariateTimeSeries`. Even if we use a struct here, we still need to covert it to an array when updating the time series.

I like Anton's idea of taking array to record the value because we are able to guarantee that, for a same time series, both series names and series values have the same dimension.

Reply:
Ok, I'm not sure about that but as seem to be discussed before, I agree.